### ISSUE TITLE
Please document the difference between StartWorkflow and ExecuteWorkflow

### ISSUE THREAD

**Issue by mingfang** [uber-go/cadence-client#866]
The current documentation for `StartWorkflow` and `ExecuteWorkflow` are very similar.
https://github.com/uber-go/cadence-client/blob/master/client/client.go#L85
https://github.com/uber-go/cadence-client/blob/master/client/client.go#L110

But base on this Slack with @mfateev, the two calls are very different and it's worth documenting the differences.
```
Ming Fang Today at 3:11 PM
I have a beginner’s questions, what is the difference between StartWorkflow and  ExecuteWorkflow; When should we use one or the other? (edited) 

Maxim  9 minutes ago
StartWorkflow starts workflow and immediately returns.
ExecuteWorkflow starts workflow and waits for it to complete.
ExecuteWorkflow is very useful for development but rarely useful in production use cases.
If you call execute workflow for already running workflow it is going to reconnect to already running workflow to wait for its completion.
The reconnection doesn't work if you specify workflowIdReusePolicy to AllowDuplicate.
```

**Comment by nturner-asp** [comment]
This is still hard to parse for people new to the codebase.  

Would be great if you could add a simple explanation on the behavioral difference, similar to what @mingfang suggested above.
